packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -1033,4 +1049,39 @@ describeCompat("IdCompressor Summaries", "NoCompat", (getTestObjectProvider) =>
 	it("Container uses short DataStore & DDS IDs in On mode", async () => {
 		await TestCompactIds("on");
 	});
+
+	it("always uses short data store IDs in detached container", async () => {
+		const loader = provider.makeTestLoader();
+		const defaultCodeDetails: IFluidCodeDetails = {
+			package: "defaultTestPackage",
+			config: {},
+		};
+		const container = await loader.createDetachedContainer(defaultCodeDetails);
+		const defaultDataStore = (await container.getEntryPoint()) as ITestFluidObject;
+		assertInvert(
+			defaultDataStore.context.id.length <= 2,
+			"Default data store's ID should be short",
+		);
+		const dataStore1 =
+			await defaultDataStore.context.containerRuntime.createDataStore(TestDataObjectType);
+		const ds1 = (await dataStore1.entryPoint.get()) as ITestFluidObject;
+		assertInvert(
```

Comment:
Should we change this test and create 100 of detached data stores?

Reply:
We could but I don't see a point. Are you suggesting we the load a container and basically validate that it can load all these data stores, process ops, etc?

Reply:
Yeah you need 13 to create a datastore with id "["

Reply:
Sure, but there is no repro of the issue even with that. So, I don't see a point in doing it yet. Once we have a repro, we can add test for this.

Reply:
Working on it

Reply:
https://github.com/microsoft/FluidFramework/pull/21625/files